From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: AltarsCN <337293323+Altars3668@users.noreply.github.com>
Date: Mon, 31 Aug 2026 13:32:42 -0400
Subject: [PATCH] ntdll: Report Windows 11 24H2 build number.

Not intended for upstream: pins the reported build to 24H2 (26100).

(cherry picked from commit c344f0b8003cad162939259b1cd962931435b13a)

diff --git a/dlls/ntdll/version.c b/dlls/ntdll/version.c
index c58fa2fe51f..6b340e6cc58 100644
--- a/dlls/ntdll/version.c
+++ b/dlls/ntdll/version.c
@@ -55,6 +55,7 @@ typedef enum
     WIN81,   /* Windows 8.1 */
     WIN10,   /* Windows 10 */
     WIN11,   /* Windows 11 */
+    WIN11_24H2, /* Windows 11 24H2 */
     NB_WINDOWS_VERSIONS
 } WINDOWS_VERSION;
 
@@ -175,6 +176,11 @@ static const RTL_OSVERSIONINFOEXW VersionData[NB_WINDOWS_VERSIONS] =
         sizeof(RTL_OSVERSIONINFOEXW), 10, 0, 22000, VER_PLATFORM_WIN32_NT,
         L"", 0, 0, VER_SUITE_SINGLEUSERTS, VER_NT_WORKSTATION, 0
     },
+    /* WIN11_24H2 */
+    {
+        sizeof(RTL_OSVERSIONINFOEXW), 10, 0, 26100, VER_PLATFORM_WIN32_NT,
+        L"", 0, 0, VER_SUITE_SINGLEUSERTS, VER_NT_WORKSTATION, 0
+    },
 };
 
 static const struct { WCHAR name[12]; WINDOWS_VERSION ver; } version_names[] =
@@ -206,6 +212,7 @@ static const struct { WCHAR name[12]; WINDOWS_VERSION ver; } version_names[] =
     { L"win81", WIN81 },
     { L"win10", WIN10 },
     { L"win11", WIN11 },
+    { L"win11_24h2", WIN11_24H2 },
 };
 
 
